src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -242,14 +283,78 @@ public override void ExecuteCmdlet()
                 openPorts: OpenPorts);
             var networkInterface = resourceGroup.CreateNetworkInterfaceConfig(
                 Name, subnet, publicIpAddress, networkSecurityGroup);
-            var virtualMachine = resourceGroup.CreateVirtualMachineConfig(
-                name: Name,
-                networkInterface: networkInterface,
-                isWindows: isWindows,
-                adminUsername: Credential.UserName,
-                adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
-                image: image,
-                size: Size);
+            ResourceConfig<VirtualMachine> virtualMachine = null;
+
+            if (image != null)
+            {
+                virtualMachine = resourceGroup.CreateVirtualMachineConfig(
+                    name: Name,
+                    networkInterface: networkInterface,
+                    isWindows: isWindows,
+                    adminUsername: Credential.UserName,
+                    adminPassword: new NetworkCredential(string.Empty, Credential.Password).Password,
+                    image: image,
+                    size: Size);
+            }
+            else
+            {
+                var storageClient =
+                        AzureSession.Instance.ClientFactory.CreateArmClient<StorageManagementClient>(DefaultProfile.DefaultContext,
+                            AzureEnvironment.Endpoint.ResourceManager);
+                var st1 = storageClient.StorageAccounts.Create(ResourceGroupName, Name, new StorageAccountCreateParameters
+                {
+#if !NETSTANDARD
+                    AccountType = AccountType.PremiumLRS,
+#else
+                    Sku = new SM.Sku
+                    {
+                        Name = SkuName.PremiumLRS
+                    },
+#endif
```

Comment:
Could we create a separate static function? For example
```c#
public static StorageAccountCreateParameters WithPremiumLRS(this StorageAccountCreateParameters p)
{
#if !NETSTANDARD 
    p.AccountType = AccountType.PremiumLRS;
#else  
    p.Sku = new SM.Sku 
    { 
          Name = SkuName.PremiumLRS  
     }; 
#endif
     return p;
}
```

Reply:
@sergey-shandar When users give a disk file, & we upload it to the cloud, there should be no lease on it. However, if users directly refer to a disk link, it's not necessarily that case, so we need to think about what's the proper way to validate that. Thus, it's not included in this PR to make things too complicated.

Reply:
Created for tracking: https://github.com/Azure/azure-powershell/issues/5279

Reply:
Regarding the macro conditionally separated code, I think we should be able to live with them for a while? Since eventually the Storage client is going to reconcile and use only the final set of input parameters. Then at that time, the static function will need to be removed as well, if they are added now.